cli/src/main/java/com/devonfw/tools/ide/url/updater/UpdateManager.java

```diff
@@ -94,14 +94,31 @@ public void updateAll() {
       if (isTimeoutExpired()) {
         break;
       }
-      try {
-        updater.setExpirationTime(getExpirationTime());
-        updater.setUrlFinalReport(this.urlFinalReport);
-        updater.update(this.urlRepository);
-      } catch (Exception e) {
-        logger.error("Failed to update {}", updater.getToolWithEdition(), e);
+      update(updater);
+    }
+  }
+
+  public void update(String tool) {
```

Comment:
Please add Javadoc or Override

Reply:
Done. Only for local dev actually and quite self-explanatory but JavaDoc is always better to have for public stuff.